Content.Server/Clothing/Systems/RestrictedClothingSystem.cs

```diff
@@ -0,0 +1,71 @@
+using Content.Server.Administration.Managers;
+using Content.Server.Atmos.EntitySystems;
+using Content.Server.Clothing.Components;
+using Content.Server.Mind.Components;
+using Content.Server.Popups;
+using Content.Server.Speech.Components;
+using Content.Shared.Administration;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Popups;
+using Robust.Shared.Player;
+
+namespace Content.Server.Clothing.Systems;
+
+public sealed class RestrictedClothingSystem : EntitySystem
+{
+    [Dependency] private readonly IAdminManager _adminManager = default!;
+    [Dependency] private readonly FlammableSystem _flammableSystem = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RestrictedClothingComponent, GotEquippedEvent>(OnGotEquipped);
+    }
+
+    private void OnGotEquipped(EntityUid uid, RestrictedClothingComponent component, GotEquippedEvent args)
+    {
+        if (!TryComp(uid, out ClothingComponent? clothing))
+            return;
+
+        // check if entity was actually used as clothing
+        // not just taken in pockets or something
+        var isCorrectSlot = clothing.Slots.HasFlag(args.SlotFlags);
+        if (!isCorrectSlot) return;
+
+        if (!TryComp(args.Equipee, out MindComponent? mind))
+            return;
+
+        if (args.Equipee.GetHashCode() == component.WhitelistedUid)
+            return;
+
+        bool applyEffect = !mind.HasMind || mind.Mind!.Session == null || component.RequireWhitelist;
+
+        if (!applyEffect)
+            foreach (var perm in component.Permissions)
+            {
+                if (!Enum.TryParse<AdminFlags>(perm, out var flag))
+                    continue;
+
+                if (_adminManager.HasAdminFlag(mind.Mind!.Session!, flag, true))
```

Comment:
Handle `Mind` and `Session` nullability

Reply:
Then you can remove the null forgiving operators, if the IDE complains there must be something wrong (`Mind` has a `TryGetSession()` method btw)